### ISSUE TITLE
FreeRDP3 Support

### ISSUE THREAD

**Issue by mac-linux-free** [asbru-cm/asbru-cm#1088]
Archlinux changed FreeRDP to FreeRDP3. Now all RDP connections do not work anymore.

Thank you

**Comment by gfrenoy** [comment]
Can you elaborate on what is not working anymore ?  Which error message, what did you try to workaround this ; any PR to suggest ?

**Comment by mac-linux-free** [comment]
The syntax has changed. Most of the options are now with slashes instead of dashes. e.g. -sec-nla +fonts or +gfx-thin-client .
I can post the new manpage of xfreerdp version 3 if you want.

**Comment by gfrenoy** [comment]
Well, if they have broken to compatibility, it means we'll need to add a new type of connection.

If someone wants to give it a try, this means (among others):
 - Add a new method in https://github.com/asbru-cm/asbru-cm/tree/master/lib/method   (xfreerdp can be used as a starting point I guess)
 - Support the new method in https://github.com/asbru-cm/asbru-cm/blob/master/lib/edit/PACMethod.pm
 - Support the new method in _getMethods of https://github.com/asbru-cm/asbru-cm/blob/master/lib/PACUtils.pm
 - Support the new method in https://github.com/asbru-cm/asbru-cm/blob/master/lib/asbru_conn

**Comment by gfrenoy** [comment]
How does this related to wlfreerdp ?  See #974.  Does FreeRDP3 supports Wayland ?  Can we merge those two requests ?

**Comment by nerijus** [comment]
As I understand FreeRDP3 is just current freerdp version, for example, Fedora 40 has freerdp-3.7.0-1.fc40.x86_64. For me asbru-cm works with connection method `RDP (xfreerdp)`, so I think this bug report is incorrect. @mac-linux-free , could it be that you are using `RDP (rdesktop)` Connection method?

**Comment by gfrenoy** [comment]
Thanks @nerijus, this is very interesting.
@mac-linux-free Can you please give us more details about the exact version of freerdp that you are using ?

**Comment by nerijus** [comment]
Could you please tell the connection method?

**Comment by ClaudeGontier** [comment]
Hello everyone,

I also encountered this problem when Arch Linux updated from FreeRDP2 to FreeRDP3.
As a workaround, I created "Generic command" entries for all my RDP connections.

Since then, Arch made FreeRDP into two packages, [freerdp](https://archlinux.org/packages/extra/x86_64/freerdp/) and [freerdp2](https://archlinux.org/packages/extra/x86_64/freerdp2/).
The default "xfreerdp" command redirected to the freerdp2 program, which fixed my issue.

I just tested now to use the FreeRDP3 binaries with Ásbrú, and it worked. Maybe the FreeRDP devs added backward compatibility for the old syntax ?
Anyways, what's important is that right now Ásbrú works with both FreeRDP2 and FreeRDP3.

**Comment by mac-linux-free** [comment]
Yes I did that also but the syntax from the options are not correct.
I got a working connection by disabling all options and add custom options ( /scale-desktop:120 /scale-device:100 -wallpaper /cert:ignore)

**Comment by mac-linux-free** [comment]
This is now my custom options line:
`
 /scale-desktop:120 /scale-device:100 +dynamic-resolution /cert:ignore /gdi:hw /gfx:progressive:on -themes -wallpaper`

but -wallpaper is not working.

Perhaps I go back to freerdp2

**Comment by yennor** [comment]
just for trying it out i've installed freerdp3 on debian testing. I've just copied the connection string asbru-cm generates and executed it in the terminal using the freerdp3 executable. seems to work. Also when looking at the man pages, they seem to be pretty similar. There are some differences in the details. 
I guess it should be pretty straight forward implementing that, by just copying the options from freerdp2 and changing the few ones which changed in freerdp3. (In the files @gfrenoy mentioned above)

**Comment by annahri** [comment]
> just for trying it out i've installed freerdp3 on debian testing. I've just copied the connection string asbru-cm generates and executed it in the terminal using the freerdp3 executable. seems to work. Also when looking at the man pages, they seem to be pretty similar. There are some differences in the details. I guess it should be pretty straight forward implementing that, by just copying the options from freerdp2 and changing the few ones which changed in freerdp3. (In the files [@gfrenoy](https://github.com/gfrenoy) mentioned above)

Adding my 2 cents, I'm having issues with mine. Probably your connection doesn't use anything that depends on freerdp2. For instance, this doesn't work on freerdp3:

```
xfreerdp3  /bpp:24  /cert-ignore -u example HOST:3389 -T "EXAMPLE"
```

Error message:
```
[776664:000bd9d8] [ERROR][com.winpr.commandline] - [CommandLineParseArgumentsA]: Failed at index 4 [<censored: build with -DWITH_DEBUG_UTILS_CMDLINE_DUMP=ON for details>]: Unexpected keyword
```

Because it uses `/cert-ignore` which in freerdp3 is changed to: `/cert:ignore`.

So far, I'm using Remmina for my RDP needs but just recently I tried to implement xfreerdp3 method in Asbru and it's working good. I'll open a PR after ensuring it has no other issues.
